core/src/main/java/org/elasticsearch/rest/action/exists/RestExistsAction.java

```diff
@@ -51,9 +58,11 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         if (RestActions.hasBodyContent(request)) {
             existsRequest.source(RestActions.getRestContent(request));
         } else {
-            QuerySourceBuilder querySourceBuilder = RestActions.parseQuerySource(request);
-            if (querySourceBuilder != null) {
-                existsRequest.source(querySourceBuilder);
+            QueryBuilder<?> queryBuilder = RestActions.parseQuerySource(request);
+            if (queryBuilder != null) {
+                QuerySourceBuilder querySourceBuilder = new QuerySourceBuilder();
+                querySourceBuilder.setQuery(queryBuilder);
+                existsRequest.source(querySourceBuilder.buildAsBytes());
```

Comment:
given that they only accept a query as body, have you considered moving over validate_query, count and exists too instead of keeping the bytes array there ? Maybe there are complications that I am not seeing though.